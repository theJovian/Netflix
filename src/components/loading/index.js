
export default function Loading({ src, ...restProps }) {
    return (
        <Spinner>
            <LockBody />
            <Picture src={`/images.users/${src}.png`} />
        </Spinner>
    )
}