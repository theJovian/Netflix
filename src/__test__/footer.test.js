import { render } from '@testing-library/react';
import { Footer } from '../components';

describe('<Footer />', () => {
	it('renders the <Footer /> with populated Data', () => {
		const { container, getByText } = render(
			<Footer>
				<Footer.Title>Question? Contact us</Footer.Title>
				<Footer.Break />
				<Footer.Row>
					<Footer.Column>
						<Footer.Link href='#'>FAQ</Footer.Link>
						<Footer.Link href='#'>Investor Relations</Footer.Link>
						<Footer.Link href='#'>Ways to Watch</Footer.Link>
						<Footer.Link href='#'>Corporate Informate</Footer.Link>
						<Footer.Link href='#'>Neflix Originals</Footer.Link>
					</Footer.Column>
					<Footer.Column>
						<Footer.Link href='#'>Help Center</Footer.Link>
						<Footer.Link href='#'>Jobs</Footer.Link>
						<Footer.Link href='#'>Terms of Use</Footer.Link>
						<Footer.Link href='#'>Contact Us</Footer.Link>
					</Footer.Column>
					<Footer.Column>
						<Footer.Link href='#'>Account</Footer.Link>
						<Footer.Link href='#'>Redeem Gift Cards</Footer.Link>
						<Footer.Link href='#'>Privacy</Footer.Link>
						<Footer.Link href='#'>Speed Test</Footer.Link>
					</Footer.Column>
					<Footer.Column>
						<Footer.Link href='#'>Media Center</Footer.Link>
						<Footer.Link href='#'>Buy Gift Cards</Footer.Link>
						<Footer.Link href='#'>Cookie Preferences</Footer.Link>
						<Footer.Link href='#'>Legal Notices</Footer.Link>
					</Footer.Column>
				</Footer.Row>
				<Footer.Break />
				<Footer.Text>Neflix United Kingdom</Footer.Text>
			</Footer>
		);

		expect(getByText('Question? Contact us')).toBeTruthy();
		expect(getByText('FAQ')).toBeTruthy();
		expect(getByText('Investor Relations')).toBeTruthy();
		expect(getByText('Corporate Informate')).toBeTruthy();
		expect(getByText('Neflix Originals')).toBeTruthy();
		expect(container.firstChild).toMatchSnapshot();
	});
});
