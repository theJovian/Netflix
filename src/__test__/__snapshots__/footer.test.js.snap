// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer /> with populated Data 1`] = `
<div
  class="footer__Container-qu6i6-0 kfmMvS"
>
  <p
    class="footer__Title-qu6i6-4 htVotQ"
  >
    Question? Contact us
  </p>
  <p
    class="footer__Break-qu6i6-6 lblvIr"
  />
  <div
    class="footer__Row-qu6i6-2 GdhWL"
  >
    <div
      class="footer__Column-qu6i6-1 canvup"
    >
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        FAQ
      </a>
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Ways to Watch
      </a>
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Corporate Informate
      </a>
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Neflix Originals
      </a>
    </div>
    <div
      class="footer__Column-qu6i6-1 canvup"
    >
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Help Center
      </a>
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Jobs
      </a>
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Terms of Use
      </a>
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Contact Us
      </a>
    </div>
    <div
      class="footer__Column-qu6i6-1 canvup"
    >
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Account
      </a>
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Redeem Gift Cards
      </a>
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Speed Test
      </a>
    </div>
    <div
      class="footer__Column-qu6i6-1 canvup"
    >
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Media Center
      </a>
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Buy Gift Cards
      </a>
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Cookie Preferences
      </a>
      <a
        class="footer__Link-qu6i6-3 krWIOC"
        href="#"
      >
        Legal Notices
      </a>
    </div>
  </div>
  <p
    class="footer__Break-qu6i6-6 lblvIr"
  />
  <p
    class="footer__Text-qu6i6-5 cSwbqj"
  >
    Neflix United Kingdom
  </p>
</div>
`;
